waiter/integration/waiter/token_watch_integration_test.clj

```diff
@@ -390,17 +395,18 @@
   (testing-using-waiter-url
     (let [{:keys [cookies]} (make-request waiter-url "/waiter-auth")
           streaming-timeout-ms 5000
-          start-time-epoch-ms (System/currentTimeMillis)
           {:keys [exit-fn go-chan headers query-state-fn]}
           (start-watch waiter-url cookies :query-params {"include" ["metadata"]
                                                          "name" "test-token-watch-streaming-timeout"
                                                          "streaming-timeout" (str streaming-timeout-ms)
                                                          "watch" "true"})
           _ (async/alts!! [go-chan (async/timeout (* 2 streaming-timeout-ms))] :priority true)
           end-time-epoch-ms (System/currentTimeMillis)
-          elapsed-time-ms (- end-time-epoch-ms start-time-epoch-ms)
+          {:keys [initial-event-time-epoch-ms token->index]} (query-state-fn)
           _ (exit-fn)
+          _ (is (some? initial-event-time-epoch-ms))
+          elapsed-time-ms (- end-time-epoch-ms initial-event-time-epoch-ms)
           assertion-message (str {:elapsed-time-ms elapsed-time-ms
                                   :headers headers})]
-      (is (empty? (query-state-fn)) assertion-message)
-      (is (<= streaming-timeout-ms elapsed-time-ms (+ streaming-timeout-ms 1000)) assertion-message))))
+      (is (empty? token->index) assertion-message)
+      (is (<= (- streaming-timeout-ms 1000) elapsed-time-ms (+ streaming-timeout-ms 1000)) assertion-message))))
```

Comment:
Why are we subtraction 1 second?

Reply:
because its now measuring the time from when the client received the initial event. That time is later than when the server sent the event because of some latency. And it's possible for that delay to be greater than the delay when the server closes the connection making the `elapsed-time-ms` slightly less than the `streaming-timeout-ms`. 

We expect the elapsed time to always be very close to the streaming-timeout.